File: push/src/main/java/com/xiaomi/mipush/sdk/ActivityLifecycleCallbacksForCR.java
package com.xiaomi.mipush.sdk;

import android.app.Activity;
import android.app.Application;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import com.xiaomi.push.service.clientReport.PushClientReportHelper;
import com.xiaomi.push.service.clientReport.PushClientReportManager;
import com.xiaomi.push.service.clientReport.ReportConstants;
import java.util.HashSet;
import java.util.Set;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/mipush/sdk/ActivityLifecycleCallbacksForCR.class */
public class ActivityLifecycleCallbacksForCR implements Application.ActivityLifecycleCallbacks {
    private Set<String> mMsgIdSet = new HashSet<>();

    private static void attachApplication(Application application) {
        application.registerActivityLifecycleCallbacks(new ActivityLifecycleCallbacksForCR());
    }

    public static void forceAttachApplication(Context context) {
        attachApplication((Application) context.getApplicationContext());
    }

    @Override // android.app.Application.ActivityLifecycleCallbacks
    public void onActivityCreated(Activity activity, Bundle bundle) {
    }

    @Override // android.app.Application.ActivityLifecycleCallbacks
    public void onActivityDestroyed(Activity activity) {
    }

    @Override // android.app.Application.ActivityLifecycleCallbacks
    public void onActivityPaused(Activity activity) {
    }

    @Override // android.app.Application.ActivityLifecycleCallbacks
    public void onActivityResumed(Activity activity) {
        Intent intent = activity.getIntent();
        if (intent == null) {
            return;
        }
        String stringExtra = intent.getStringExtra("messageId");
        int intExtra = intent.getIntExtra(ReportConstants.EVENT_MESSAGE_TYPE, -1);
        if (TextUtils.isEmpty(stringExtra) || intExtra <= 0 || this.mMsgIdSet.contains(stringExtra)) {
            return;
        }
        this.mMsgIdSet.add(stringExtra);
        if (intExtra == 3000) {
            PushClientReportManager.getInstance(activity.getApplicationContext()).reportEvent(activity.getPackageName(), PushClientReportHelper.getInterfaceIdByType(intExtra), stringExtra, ReportConstants.AWAKE_TYPE_APP_START, null);
        } else if (intExtra == 1000) {
            PushClientReportManager.getInstance(activity.getApplicationContext()).reportEvent(activity.getPackageName(), PushClientReportHelper.getInterfaceIdByType(intExtra), stringExtra, 1008, null);
        }
    }

    @Override // android.app.Application.ActivityLifecycleCallbacks
    public void onActivitySaveInstanceState(Activity activity, Bundle bundle) {
    }

    @Override // android.app.Application.ActivityLifecycleCallbacks
    public void onActivityStarted(Activity activity) {
    }

    @Override // android.app.Application.ActivityLifecycleCallbacks
    public void onActivityStopped(Activity activity) {
    }
}

File: legacy/src/main/java/com/xiaomi/mipush/sdk/ActivityLifecycleCallbacksForCR.kt
package com.xiaomi.mipush.sdk

import android.app.Activity
import android.app.Application
import android.content.Context
import android.os.Bundle
import android.text.TextUtils
import com.xiaomi.push.service.clientReport.PushClientReportHelper
import com.xiaomi.push.service.clientReport.PushClientReportManager
import com.xiaomi.push.service.clientReport.ReportConstants

/*
 * Current override reference: com.xiaomi.xmsf 0.3.17-20260410000745 (versionCode 1003003000),
 * base.apk sha256 f3d72b6f5e1427ceecd3147a051d58e4dc95bb528397d486658e01cad9f7e590,
 * JADX path: com.xiaomi.xmsf/current/base/sources/com/xiaomi/mipush/sdk/ActivityLifecycleCallbacksForCR.java
 * No stock 7.4.67-C same-path source was found in the split source tree.
 */
class ActivityLifecycleCallbacksForCR : Application.ActivityLifecycleCallbacks {
    private val mMsgIdSet: MutableSet<String> = HashSet()

    override fun onActivityCreated(activity: Activity, bundle: Bundle?) {}

    override fun onActivityDestroyed(activity: Activity) {}

    override fun onActivityPaused(activity: Activity) {}

    override fun onActivityResumed(activity: Activity) {
        val intent = activity.intent ?: return
        val messageId = intent.getStringExtra("messageId")
        val eventMessageType = intent.getIntExtra(ReportConstants.EVENT_MESSAGE_TYPE, -1)
        if (TextUtils.isEmpty(messageId) || eventMessageType <= 0 || mMsgIdSet.contains(messageId)) {
            return
        }
        mMsgIdSet.add(messageId ?: "")
        if (eventMessageType == ReportConstants.AWAKE_TYPE) {
            PushClientReportManager.getInstance(activity.applicationContext).reportEvent(
                activity.packageName,
                PushClientReportHelper.getInterfaceIdByType(eventMessageType),
                messageId ?: "",
                ReportConstants.AWAKE_TYPE_APP_START,
                null
            )
        } else if (eventMessageType == ReportConstants.NOTIFICATION_TYPE) {
            PushClientReportManager.getInstance(activity.applicationContext).reportEvent(
                activity.packageName,
                PushClientReportHelper.getInterfaceIdByType(eventMessageType),
                messageId ?: "",
                ReportConstants.NOTIFICATION_TYPE_APP_START,
                null
            )
        }
    }

    override fun onActivitySaveInstanceState(activity: Activity, bundle: Bundle) {}

    override fun onActivityStarted(activity: Activity) {}

    override fun onActivityStopped(activity: Activity) {}

    companion object {
        private fun attachApplication(application: Application) {
            application.registerActivityLifecycleCallbacks(ActivityLifecycleCallbacksForCR())
        }

        @JvmStatic
        fun forceAttachApplication(context: Context) {
            attachApplication(context.applicationContext as Application)
        }
    }
}
